### DISCUSSION TITLE
curl to add a comment to PR is not working

### DISCUSSION THREAD

**Discussion by elangosenthilnathan** [community/community#70062] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a requirement where I need to add a custom content into a PR as a comment , this has to be done within the workflow actions. I have this code, however this does not seem to work. I do not want to associate the comment with any file or commit. Is there a way to add a generic comment.

On postman, I was able to submit a comment just with the "body" element in json-data. 

      - name: Add comment to PR
        shell: bash
        run: |
          BESTFIX_OUTPUT="Sample test content"
          git_hash=$(git rev-parse "$GITHUB_SHA")
  
          JSON_STRING="{
             \"body\":\"${BESTFIX_OUTPUT}\",
             \"commit_id\":\"${git_hash}\",
             \"path\":\"file.yml\",
             \"line\":2,
             \"start_line\":1,
             \"start_side\":\"RIGHT\",
             \"side\":\"RIGHT\"
          }"       
          
          url="https://api.github.com/repos/${{ github.repository_owner }}/${{ github.event.repository.name }}/pulls/${{github.event.number}}/comments"

          curl -L \
            -X POST \
            -H 'Content-type:application/json' \
            -H "Authorization: Bearer ${GH_API_TOKEN}" \
            "${url}" \
            -d "$JSON_STRING"
        env:
          GH_API_TOKEN: ${{ secrets.API_TOKEN }}
          SHIFTLEFT_ACCESS_TOKEN: ${{ secrets.SHIFTLEFT_ACCESS_TOKEN }}  

I tried to follow this page https://docs.github.com/en/rest/pulls/comments?apiVersion=2022-11-28#create-a-review-comment-for-a-pull-request

**Comment by JairTorres1003** [comment]
Hi @elangosenthilnathan

If you just want to add a comment to the pull directly, you could do the following.

```yml
    - name: Add comment to PR
      run: |
        # The comment content
        BESTFIX_OUTPUT="Sample test content"
        
        # Use jq to format the comment as JSON
        comment_json=$(jq -n --arg body "$BESTFIX_OUTPUT" '{"body": $body'})

        url="https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.pull_request.number }}/comments"

        curl -L \
          -X POST \
          -H 'Content-type: application/json' \
          -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
          "${url}" \
          -d "$comment_json"
```

Use `GITHUB_TOKEN` is the default token that provides the necessary permissions to make the comment.

If you like to dig a little deeper, I have designed a workflow that adds a comment to the pulls based on a search for a specific word

https://github.com/JairTorres1003/JT-Workflows/blob/main/workflows/file-checker.yml#L48-L66

**Comment by elangosenthilnathan** [comment]
Thank You @JairTorres1003 Yes I figured out that every PR is an issue and adding a comment to the issue api is easier.

**Reply by JairTorres1003** [reply, marked as answer]
You're welcome, you can now mark the answer as accepted to indicate that this question has already been answered.
